pkg/cmd/render/render.go

```diff
@@ -109,98 +134,218 @@ func runRenderCmd(cmd *cobra.Command, args []string) {
-		podPath := filepath.Join(ccoRenderDir, bootstrapManifestsDir, "cloud-credential-operator-pod.yaml")
+		podPath := filepath.Join(ccoRenderDir, bootstrapManifestsDir, podYamlFilename)
 		podContent := fmt.Sprintf(podTemplate, renderOpts.ccoImage)
 		log.Infof("writing file: %s", podPath)
-		err = ioutil.WriteFile(podPath, []byte(podContent), 0644)
+		err := ioutil.WriteFile(podPath, []byte(podContent), 0644)
 		if err != nil {
-			log.WithError(err).Fatal("failed to write file")
+			return errors.Wrap(err, "failed to write file")
 		}
 	} else {
 		log.Info("CCO disabled, skipping static pod manifest.")
 	}
+
+	return nil
+}
+
+func writeFile(filePath string, fileData []byte) error {
+	log.Infof("Writing file: %s", filePath)
+	err := ioutil.WriteFile(filePath, fileData, 0644)
+	if err != nil {
+		return errors.Wrap(err, "failed to write file")
+	}
+	return nil
+}
+
+func isValidMode(mode string) bool {
+	switch mode {
+	case string(operatorv1.CloudCredentialsModeManual):
+	case string(operatorv1.CloudCredentialsModeMint):
+	case string(operatorv1.CloudCredentialsModePassthrough):
+	case "":
+		// all the above are acceptable
+	default:
+		return false
+	}
+	return true
 }
 
 // isDisabled will search through all the files in destinationDir (which also contains
-// the source manifests) for a configmap indicating whether or not CCO is disabled. In
+// the source manifests) for the deprecated configmap indicating whether or not CCO is disabled. In
 // the absence of any configmap, it will return the default disabled setting (which is
 // that the operator is enabled by default) for the operator.
-func isDisabled() bool {
+func isDisabledViaConfigmap() bool {
 
 	// if were were not provided a place to search for the install-time manifests,
 	// just return the default CCO operator enabled/disabled value
 	if renderOpts.manifestsDir == "" {
 		return utils.OperatorDisabledDefault
 	}
 
-	files, err := ioutil.ReadDir(renderOpts.manifestsDir)
+	cm, err := getConfigMap(renderOpts.manifestsDir, minterv1.CloudCredOperatorNamespace, constants.CloudCredOperatorConfigMap)
 	if err != nil {
-		log.WithError(err).Errorf("failed to list files in %s, using defualt operator settings", renderOpts.destinationDir)
+		log.WithError(err).Warnf("errored while searching for ConfigMap %s/%s, using defaults", minterv1.CloudCredOperatorNamespace, constants.CloudCredOperatorConfigMap)
 		return utils.OperatorDisabledDefault
 	}
+	if cm == nil {
+		log.Debugf("configmap %s/%s not found, using default mode", minterv1.CloudCredOperatorNamespace, constants.CloudCredOperatorConfigMap)
+		return utils.OperatorDisabledDefault
+	}
+	logger := log.New()
+	logger.SetLevel(log.GetLevel())
+	disabled, err := utils.CCODisabledCheck(cm, logger)
+	if err != nil {
+		return utils.OperatorDisabledDefault
+	}
+
+	return disabled
+}
+
+type basicInstallConfig struct {
+	CredentialsMode string `yaml:"credentialsMode"`
+}
+
+func getModeFromInstallConfig() (string, error) {
```

Comment:
Change this to return an `operatorv1.CloudCredentialsMode` so that you don't have to have casts littered throughout the code elsewhere.